### ISSUE TITLE
may 12

### ISSUE THREAD

**Issue by hillarysanders** [hillarysanders/notes#31]
this week:
looked into CONVNET model decay over time, and worked on tracking the learning curve for CONVNET. 
yesterday:
- yesterday was mostly reading that verizon paper, and working out kinks in code - ran the:
 - sensitivity analysis for malware definition, and also tracked dome detection rates at fixed FPRs in addition just AUC values.
  - compared validatoin data w/ and w/out URL duplicates.
  - So, the results... Basically, it looks like our models aren't super upset by new URLs. Their accuracy always goes down a bit, but they still do well. However, they do start crumpling when certain domains / patterns that were just under our labelling cutoff in the training set have now exceeded our labeling cutoff in the test set. e.g. our g.mazis.ru domain. As a result, if the test labelling definition is more extreme than our train definition (e.g. 10-0 test 5-0 train), the models have very little accuracy decay. However, if the test definition is the same, our models decay rapidly and significantly. This holds true for detection rates at FPR=10^-3,4,5, as well as AUC.

OK, so what does that mean for us?
1) We've got to work on the way we decide on our labels. If there's a better way to do it, we should be using that. It does feel like beating a dead horse with a stick... But this dead horse is actually really quite important to beat dead, sooo? :)
2)  We could consider thinking about how we want to update our models with this new information in production. Because we will be able to guesstimate what's going to hurt us by figuring out what new e.g. domains have recently been determined to be malicious by vendors on VT. How often will our models be updated with new data during production? Continuously just with e.g. daily staged testing?
Hmmm...

- [x] pester IT about when you can get your computer re-imaged - note that you need to get data access soon so you can start on an analysis for BlackHat.
- [x] now that you can access Sophos hub, go through the new hire checklist
- [ ] Start thinking about blackhat talk. What is your 3rd data source going to be? You want Sophos ben / mal, VT ben / mal, and what else?
- [x] start off model that uses 3.5m observations.
- [ ] write code to handle sophos data? I want to test AND train on it...
- [ ] grab all incorrect classifications for a time window. Grab the domains for each of these. For each domain, 1) did it simply not exist in our training dataset, or 2) what % of labels were 0 / 1 in our training dataset (compare to test dataset). In other words - how often are we seeing new domains, new content, vs how often are VT labels changing with regards to single domains. And while we're on the topic - Do domain == URL classifications for the most part? How much noise is there?

**Comment by hillarysanders** [comment]
fishtank 
urlquery.net
etc - ask Rich
# todo: add Sophos URLs to the data to be tested on the deep ec2 instance. Also train on Sophos URLs as well.
